samza-core/src/test/java/org/apache/samza/clustermanager/TestContainerPlacementActions.java

```diff
@@ -528,6 +528,124 @@ public Void answer(InvocationOnMock invocation) {
     assertFalse(containerPlacementMetadataStore.readContainerPlacementRequestMessage(requestMessage.getUuid()).isPresent());
   }
 
+
+  @Test(timeout = 20000)
```

Comment:
This test is littered with thread.sleep and timeouts -- a typical recipe for flaky jarvis/test, and increasing build times.
Would it be possible to reduce/consolidate them?

Reply:
I have used semaphores where it was easy to inject (awaiting container launch), this timeout configured on tests is generous enough, each callback spawns a different thread to mimic AMRMClientAsync behavior, so if I were to do all of these with using semaphores instead of thread.sleep for state change, I would need some significant refactors

Reply:
These are integ tests and it takes 35 seconds to run all of them serially, we can also make them run in parallel to further reduce build times